### ISSUE TITLE
Unwanted grandchildren are being created on mobile

### ISSUE THREAD

**Issue by githubbob42** [githubbob42/mingle2github2#5439]
**[Mingle Card: 5800](https://liquidframeworks.mingle.thoughtworks.com/projects/alpine_mobile/cards/5800)**
Steps to Reproduce
------------------

```
| | |
|-|-|
|**Version #**|4.0.1.4033 & 4.0.1.4035|
|**Hardware**| |
|**OS**| |
|**Browser**| |
|**Username**| |
|**Password**|Use LastPass|
|ORG ID|00D1a000000KXoU|
|User ID| |
|RayGun Error ID| |
  
 

```

1.  Org needs to have parent and child pricebook items
2.  In mobile, create a ticket and add a parent ticket item
3.  select parent and add a child ticket item (click gear, click add child items)
4.  in the ticket item search screen, perform a search that will end in one ticket item as the result
5.  Quick Add the ticket item (you should still be on the ticket item search screen after the add is complete)
6.  Stay in the TI search and add another ticket item

Expected Result
---------------

_Single Ticket Items should be added to the ticket item list, any related ticket reports that generate should show only the expected ticket items, the number of events that show on the mobile home page should reflect the correct number of ticket item adds._

_(Not sure if this needs to be a separate ticket or not) The Ticket Item Menu should also display the appropriate menu options. If a parent TI is selected, the add child items options should be an option, if you click a parent and a child, the Add Child Items options should not be available. However, after Step 6, if you close the Search page, and you unselect the chidren ticket items, so only the parent item is left selected, then the Add Child Items is not available on the Gear menu. There selected item count is off at this point which indicates there’s a ‘hidden’ extra item that was created and selected._

Actual Result
-------------

_Ticket Item list in mobile only shows the items you added, however, there are extra events that show in dev tools and on the pending changes on mobile home page. Step 6 is what generates the unwanted grandchildren. In Dev Tools you can see that the parent ticket item ID is the syncID of the previous child ticket item created in Step 5._

_Analysis_
----------

Related Cards
-------------

Card #5772

Steps for Creating a Defect Card
--------------------------------

```
| | |
|-|-|
 
  
   
   

1

   |
   
   

Ensure the defect title and description is clear and understandable.

   |
   
   

   |
  
   
   

2.

   |
   
   

Ensure the following are listed on the card:

   

    *   Mobile or back office version.
    *   Operating system
    *   Devices
    *   Browsers
    *   Username/Password.
   

   |
   
   

   |
  
   
   

3. 

   |
   
   

Ensure there are steps to reproduce and are easy to follow.

   

Add screenshots as necessary for clarity

   |
   
   

   |
  
   
   

4. 

   |
   
   

Ensure the Expected and Actual results are listed.

   |
   
   

   |
  
   
   

5.

   |
   
   

Check whether the bug exists in production (Sync V4) and/or Sync V4 Beta

   

    *   If the bug exists in current production then select the “**Sync V4 Channel”**
    *   If the bug exists in the Beta Channel but is not in production yet, then select “**Beta Channel”**
    *   If the bug was created during current iteration then select "**Regression**”
   

   |
   
   

   

   

   

   

   

   

   

   |
  
 

```

Test Plan
---------

```
| | |
|-|-|
 
  
   
   

1.

   |
   
   

Ensure the card has enough information from the programmer before you start the verification

   

If not request more information

   |
   
   

   |
  
   
   

2.

   |
   
   

Ensure you’re able to reproduce the defect prior to verifying it

   |
   
   

   |
  
   
   

3.

   |
   
   

Ensure to verify if the PR is still valid by going to Github.

   |
   
   

   |
  
   
   

3.

   |
   
   

Create a test plan and write/update test case for the card is there is no test case in Tarantula.

   |
   
   

   |
  
   
   

4

   |
   
   

Test the card on all required devices and versions. If it’s a mobile card, always test the offline functionality around that defect. Attach screenshots to the card as necessary displaying the fix

   |
   
   

   |
  
   
   

5.

   |
   
   

Add the following test result documentations:

   

    *   Test Status:
    *   PR Build:
    *   Username/Password
    *   Test case name:
    *   Environment and devices tested on:
    *   Test Note.
   

   |
   
   

   |
  
   
   

6.

   |
   
   

Push the card to “Testing Complete”

   |
   
   

   |
  
 

```

1.  select a parent ticket item
2.  open ticket item picker using “add child item”
3.  add multiple child items using consecutive quick add operations
4.  close ticket item picker
5.  verify “selected items” count on the “gear” button is correct
6.  verify “add child item” option remains available
7.  verify extra items don’t show up in reports

Selenium > dynamicChildren.js > Test 1. Dynamic Parenting - Add children to an existing ticket item without details (#5800)

**Comment by githubbob42** [comment]
**Pull Request #2757**
